fastlane/spec/actions_specs/carthage_spec.rb

```diff
@@ -666,6 +688,55 @@
           end
         end
       end
+
+      context "when specify archive" do
+        context "when command is archive" do
+          let(:command) { 'archive' }
+
+          it "raises an exception" do
+            expect do
+              Fastlane::FastFile.new.parse("lane :test do
+                  carthage(command: '#{command}', archive: true)
+                end").runner.execute(:test)
+            end.to raise_error("Archive option is available only for 'build' command.")
+          end
+        end
+
+        context "when command is update" do
+          let(:command) { 'update' }
+
+          it "raises an exception" do
+            expect do
+              Fastlane::FastFile.new.parse("lane :test do
+                  carthage(command: '#{command}', archive: true)
+                end").runner.execute(:test)
+            end.to raise_error("Archive option is available only for 'build' command.")
+          end
+        end
```

Comment:
Thank you for adding more coverage on this validation but this check `Archive option is available only for 'build' command` is already covered by L728-L736.  Aren't these a bit redundant? It's nice to have for a short term but might be getting harder to maintain this spec file by the number of assertions or just lines.

Would you mind if you remove pointed lines?

Reply:
Or please combine the same assertions into one as I commented here?  

https://github.com/fastlane/fastlane/pull/18090/files#r569146277

Reply:
Thanks for your review!
I will combine test cases in one case.